src/ngraph/pass/manager.cpp

```diff
@@ -98,6 +98,13 @@ void pass::Manager::run_passes(shared_ptr<Function> func, bool transitive)
         {
             for (shared_ptr<Function> f : fs)
             {
+                // This checks is to skip the graph optimization when the graph pass relies on static shape
+                // but the function state is dynamic
+                if (function_pass->get_property(pass::PassProperty::REQUIRE_STATIC_SHAPE) &&
+                    f->is_dynamic())
```

Comment:
I was thinking if we can store a cached `bool` value with the `fs` so we don't need to call `f->is_dynamic()` frequently, and update the cached `bool` value like you did for the `GraphRewrite` matchers.